From 3005ea6ba96999944ea7b5cf8fabc2119dfe958a Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@renesas.com>
Date: Mon, 20 Jun 2022 16:28:52 +0700
Subject: [PATCH 02/27] pinctrl: pinctrl-rzg2l: change irq_disable to
 irq_shutdown function

This commit is created to change irq_disable to irq_shutdown function.
Currently, we used a pair of irq_unmask/irq_mask to control GPIO IRQ.
So that using irq_disable is not good. It should be changed to
irq_shutdown.
irq_shutdown will be called when any drivers free its GPIO IRQ.

Signed-off-by: hienhuynh <hien.huynh.px@renesas.com>
---
 drivers/pinctrl/renesas/pinctrl-rzg2l.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pinctrl/renesas/pinctrl-rzg2l.c b/drivers/pinctrl/renesas/pinctrl-rzg2l.c
index 4c911b5942f6..b80b64695a15 100644
--- a/drivers/pinctrl/renesas/pinctrl-rzg2l.c
+++ b/drivers/pinctrl/renesas/pinctrl-rzg2l.c
@@ -972,7 +972,7 @@ static int rzg2l_gpio_irq_check_tint_slot(struct rzg2l_pinctrl *pctrl,
 	return i;
 }
 
-static void rzg2l_gpio_irq_disable(struct irq_data *d)
+static void rzg2l_gpio_irq_shutdown(struct irq_data *d)
 {
 	struct gpio_chip *chip = irq_data_get_irq_chip_data(d);
 	struct rzg2l_pinctrl *pctrl = gpiochip_get_data(chip);
@@ -1649,7 +1649,7 @@ static int rzg2l_gpio_register(struct rzg2l_pinctrl *pctrl)
 	dev_dbg(pctrl->dev, "Registered gpio controller\n");
 
 	irq_chip->name = dev_name(pctrl->dev);
-	irq_chip->irq_disable = rzg2l_gpio_irq_disable;
+	irq_chip->irq_shutdown = rzg2l_gpio_irq_shutdown;
 	irq_chip->irq_mask = rzg2l_gpio_irq_mask;
 	irq_chip->irq_unmask = rzg2l_gpio_irq_unmask;
 	irq_chip->irq_set_type = rzg2l_gpio_irq_set_type;
-- 
2.17.1

